tests/rptest/tests/datalake/query_engine_base.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2024 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from abc import ABC, abstractmethod
+from contextlib import contextmanager
+from enum import Enum
+
+
+class QueryEngineType(str, Enum):
+    SPARK = 'spark'
+    TRINO = 'trino'
+
+
+class QueryEngineBase(ABC):
+    """Captures all the common operations across registered query engines"""
+    @staticmethod
+    @abstractmethod
+    def engine_name() -> QueryEngineType:
+        raise NotImplementedError
+
+    @abstractmethod
+    def make_client(self):
+        raise NotImplementedError
+
+    @contextmanager
+    def run_query(self, query):
+        client = self.make_client()
+        assert client
+        try:
+            try:
+                cursor = client.cursor()
+                cursor.execute(query)
+                yield cursor
+            finally:
+                cursor.close()
+        finally:
+            client.close()
+
+    def run_ddl(self, query):
```

Comment:
nit: why not just call this `run_query_all()` or somesuch? It seems useful for more than ddl, and can be useful for dml too, e.g. if we know the table is small, just list the table. Or run queries with limited number of return values

Reply:
sg renaming to run_query_fetch_all()